include/dlaf/eigensolver/gen_to_std/mc/gen_to_std_L.h

```diff
@@ -0,0 +1,120 @@
+#include "dlaf/util_matrix.h"
+
+namespace dlaf {
+namespace internal {
+namespace mc {
+
+// Implementation based on LAPACK Algorithm for the transformation from generalized to standard
+// eigenproblem (xHEGST)
+
+// Local implementation
+template <class T>
+void genToStd_L(Matrix<T, Device::CPU>& mat_a, Matrix<T, Device::CPU>& mat_l) {
+  constexpr auto Right = blas::Side::Right;
+  constexpr auto Left = blas::Side::Left;
+  constexpr auto Lower = blas::Uplo::Lower;
+  constexpr auto NonUnit = blas::Diag::NonUnit;
+  constexpr auto NoTrans = blas::Op::NoTrans;
+  constexpr auto ConjTrans = blas::Op::ConjTrans;
+
+  using hpx::util::unwrapping;
+
+  using hpx::threads::executors::pool_executor;
+  using hpx::threads::thread_priority_high;
+  using hpx::threads::thread_priority_default;
+
+  // Set up executor on the default queue with high priority.
+  pool_executor executor_hp("default", thread_priority_high);
+
+  // Set up executor on the default queue with default priority.
+  pool_executor executor_normal("default", thread_priority_default);
+
+  const SizeType m = mat_a.nrTiles().rows();
+  const SizeType n = mat_a.nrTiles().cols();
+
+  for (SizeType k = 0; k < n; ++k) {
+    const auto kk = LocalTileIndex{k, k};
+
+    // Direct transformation to standard eigenvalue problem of the diagonal tile
+    hpx::dataflow(executor_hp, unwrapping(tile::hegst<T, Device::CPU>), 1, Lower, std::move(mat_a(kk)),
+                  std::move(mat_l(kk)));
+
+    if (k != (n - 1)) {
+      for (SizeType i = k + 1; i < m; ++i) {
+        // Working on panel...
+        const auto ik = LocalTileIndex{i, k};
+        hpx::dataflow(executor_normal, unwrapping(tile::trsm<T, Device::CPU>), Right, Lower, ConjTrans,
+                      NonUnit, 1.0, mat_l.read(kk), std::move(mat_a(ik)));
+        hpx::dataflow(executor_normal, unwrapping(tile::hemm<T, Device::CPU>), Right, Lower, -0.5,
+                      mat_a.read(kk), mat_l.read(ik), 1.0, std::move(mat_a(ik)));
+      }
+
+      for (SizeType j = k + 1; j < n; ++j) {
+        // Working on trailing matrix...
+        const auto jj = LocalTileIndex{j, j};
+        const auto jk = LocalTileIndex{j, k};
+        hpx::dataflow(executor_hp, unwrapping(tile::her2k<T, Device::CPU>), Lower, NoTrans, -1.0,
+                      mat_a.read(jk), mat_l.read(jk), 1.0, std::move(mat_a(jj)));
+
+        for (SizeType i = j + 1; i < m; ++i) {
+          const auto ik = LocalTileIndex{i, k};
+          const auto ij = LocalTileIndex{i, j};
+          hpx::dataflow(executor_normal, unwrapping(tile::gemm<T, Device::CPU>), NoTrans, ConjTrans,
+                        -1.0, mat_a.read(ik), mat_l.read(jk), 1.0, std::move(mat_a(ij)));
+          hpx::dataflow(executor_normal, unwrapping(tile::gemm<T, Device::CPU>), NoTrans, ConjTrans,
+                        -1.0, mat_l.read(ik), mat_a.read(jk), 1.0, std::move(mat_a(ij)));
+        }
+      }
+
+      for (SizeType i = k + 1; i < m; ++i) {
```

Comment:
Here you can easily use `common::iterate_range2d`

Reply:
@albestro would you mind checking if I implemented it correctly, please? Thanks!